docs/source/nutshell.rst

```diff
@@ -0,0 +1,223 @@
+The value 42 is the actual parameter which  replaces the formal parameter ``x`` in the  output script :code:`fun(x) . x == 42`,  and makes the script evaluate to true.
+Every other witness would make the script evaluate to false,
+and would prevent the transaction ``T1`` to  be added to the blockchain. 
+In Bitcoin, a transaction cannot be spent twice:
+hence, once ``T1`` is on the blockchain,
+no other transaction having ``T`` as input can be appended.
+
+Note that ``T1`` is redeeming exactly the ``50 BTC`` deposited in ``T``:
+in practice, to be able to append ``T1`` to the blockchain,
+the value in output of a transaction must be strictly less
+than the value in input.
+The difference is retained by Bitcoin miners as a fee for their work.
+Currently, transactions with zero fee are not likely to be added to the blockchain. 
+
+Now, let us insert both ``T`` and ``T1`` in the editor.  While we
+write, the editor performs some static checks and signals the
+erorrs. For instance, if instead of value ``42`` we provide another
+witness for ``T``, (say for instance value ``4``) the editor will
+display a warning. If the input field of ``T1`` has a wrong reference
+(say ``T3``), or if the total amount of outgoing bitcoin is greater
+than the incoming one, the editor will signal the error.
+
+.. _label_t_signature_modeling:
+
+"""""""""""""""""""""""""""""""
+Signature verification 
+"""""""""""""""""""""""""""""""
+
+The output script of ``T`` is naïve, since anyone can produce the right witness.
+Usually, one wants to transfer bitcoins to a specific user.
+For instance, the following piece of code makes the 50 BTC in ``T1``
+redeemable only by user Alice: 
+
+.. code-block:: btm
+
+	const addrA = wif:mvMKu24MRsSDsu6LatViZtx64J6Cxsz3qe // Alice's address
+
+	transaction T1 {
+		input = T: 42
+		output = 50 BTC: fun(x) . versig(addrA; x)
+	}
+
+
+The constant ``addrA`` declares Alice's *address*:
+basically, it is the hash of Alice's public key.
+The address is prefixed by the keyword ``wif:``,
+which means that it is represented in the
+:doc:`Wallet Import Format types <types>` [#f1]_.
+Note that users may generate as many addresses as they want.
+
+The :ref:`predicate <label_c_functions>` ``versig(addrA; x)``
+in the output field of ``T1`` 
+evaluates to true if ``x`` is a valid signature
+of the transaction which redeems ``T1``,
+done with Alice's private key. 
+
+The transaction ``T1`` can be redeemed by a transaction ``T2`` made as follows:
+
+.. code-block:: btm
+
+	const skA = wif:cMpBA2wEGKrN1vpxSBzWWzFBwnrypR4rBev4vTbEDPHMeSMyyyMi // Alice's private key
+
+	transaction T2 {
+		input = T1: sig(skA)
+		output = 50 BTC: ... // any condition chosen by Alice
+	}
+
+The witness ``sig(skA)`` is the :ref:`signature <label_c_functions>`
+of transaction ``T2`` (without considering the witness itself)
+using the private key ``skA``.
+
+To perform examples with Bitcoin addresses and keys, one can use the online key generator in
```

Comment:
To perform examples with Bitcoin addresses and keys, one can use the online key generator in the sidebar of the editor page. ->
One can use the online form on the sidebar to generate new keys and addresses.

Reply:
ok